### DISCUSSION TITLE
How to get the array of filtered rows array in QTable?

### DISCUSSION THREAD

**Discussion by meo1** [quasarframework/quasar#13871] [General - Components / Directives / etc]
I am trying to solve the problem in the title and reading the documentation, the documentation refers twice to source code but where is exactly the source code for filtering issues?

![image](https://user-images.githubusercontent.com/35503088/177279503-8c82198e-f72c-4810-be99-e756ac2705e4.png)

![image](https://user-images.githubusercontent.com/35503088/177282266-d0840ebb-bf52-4802-bada-6b1c0201981c.png)

**Comment by MilosPaunovic** [comment]
[https://codepen.io/milospaunovic/pen/QWmjxNx?editors=1011](https://codepen.io/milospaunovic/pen/QWmjxNx?editors=1011)

**Reply by meo1** [reply]
Your reply is not a answer for my question. But thanks any way.

**Reply by meo1** [reply]
Or your codepen code is not complete yet.

**Reply by meo1** [reply]
I am sorry, I did not read your code carefully, it is related with my question but it does not return filtered array but the whole array.

**Reply by MilosPaunovic** [reply]
You need to filter it in the handler method however you want and return the filtered array.

**Reply by meo1** [reply]
I am sorry Milos, I cannot understand you completely. I am trying to get list of rows in array which are filtered by Quasar's default filtered, I do not want to custom filter for my code. But I checked whether I can filter in the custom function, but there is not filter method defined for QTable.

**Comment by MilosPaunovic** [comment]
[https://codepen.io/milospaunovic/pen/xxWwJWP?editors=1010](https://codepen.io/milospaunovic/pen/xxWwJWP?editors=1010)

**Reply by meo1** [reply]
Thank you, it works.

**Reply by meo1** [reply]
I am sorry, at first glance I said OK this is what I wanted to do. But is not, because it filters also according to rows in page.

**Reply by MilosPaunovic** [reply]
I don't understand what you mean by `filters according to rows in page`.

**Reply by meo1** [reply]
For example if your table renders only 5 records per page, your variable " getComputedRows" can contain max. 10 records, if your table renders only 5 records per page, your variable " getComputedRows" can contain max. 10 records. But I want all the records filtered by filter criteria.

**Reply by yusufkandemir** [reply]
@meo1 then you need `filteredSortedRows`, not `computedRows`. https://github.com/quasarframework/quasar/discussions/13871#discussioncomment-3083884

**Reply by MilosPaunovic** [reply]
https://codepen.io/milospaunovic/pen/xxWwJWP?editors=1010

**Reply by yusufkandemir** [reply]
@MilosPaunovic this would be simpler IMO:

```js
const tableRef = ref();
const filteredSortedRows = computed(() => tableRef.value?.filteredSortedRows);
```

**Reply by MilosPaunovic** [reply]
Completely agree. Above was the product of few iterations until I figured out what was asked intially.

**Reply by meo1** [reply]
Hello Milos, hello Yusuf I thank you your immediate responses. But there are some errors. First here there is a type error:

https://codepen.io/milospaunovic/pen/xxWwJWP?editors=1010

It was written `return {
      getComputedRows: computed(() => {
        const helper = { ...table.value };
        return helper?.filteredSortedRow;
      }),`

It should be "filteredSortedRows", then it worked.

Second I liked the simpler code that Yusuf suggested but it did not worked by me. Can you open codepen page for it.

**Reply by yusufkandemir** [reply]
@meo1 my snippet is supposed to work as-is on <script setup>, but with setup(), you would need to return the variables too. Also, you would need to connect it like `<q-table ref="tableRef">` as I used a different variable name for it. Don't forget to reference the correct variable name too, e.g. `{{ filteredSortedRows }}`.

So, in Codepen it would look like this:
```js
const { ref, computed } = Vue;

const app = Vue.createApp({
  setup() {
    const tableRef = ref();
    const filteredSortedRows = computed(() => tableRef.value?.filteredSortedRows);

    return {
      tableRef,
      filteredSortedRows,
// ...
```

**Reply by meo1** [reply]
Hello, this is me again, your solutions work but if I want to store the array of 'filteredSortedRows' and want to use on some other pages, I experience problems. Do you have any idea, what might it cause?

**Comment by yusufkandemir** [comment]
For getting the array of filtered rows from a QTable instance:
```js
// <q-table ref="tableRef" ... />
const tableRef = ref();

// Access what's suitable for you
tableRef.value.filteredSortedRows // filtered, sorted
tableRef.value.computedRows // filtered, sorted, paginated
```

List of available public fields, we'll document them on the docs and generate TS definitions for them soon:
https://github.com/quasarframework/quasar/blob/00d0a89ef7d022b02923c86cef3961c3f1a1fead/ui/src/components/table/QTable.js#L1019-L1023

It's not very clear what you want to do with the array of filtered rows. `filterMethod` prop is irrelevant in your case as it's used for filtering the array yourself, not getting the filtered array, but I will share it anyways. Steps to find the source code via GitHub UI:
1. Search it (https://github.com/quasarframework/quasar/search?q=filterMethod)
![image](https://user-images.githubusercontent.com/6266078/177308520-b39950ed-d378-4ccb-9834-1032d68ab800.png)
2. Click on the related file.
3. There you have it:
https://github.com/quasarframework/quasar/blob/78df1f4939936fdbf591d90addf950a8f42e665b/ui/src/components/table/table-filter.js#L12-L20
